Detectors/AOD/src/StandaloneAODProducer.cxx

```diff
@@ -0,0 +1,169 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \brief Produce AOD from reconstructed objects
+
+/// This is a standalone demonstrator producer of the AOD format
+/// from MC-RECO. For the moment not a DPL device but in future this
+/// promotion is targeted. The executable produces the aod.root file which
+/// can be fed into analysis workflows.
+
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/TableBuilder.h"
+#include "Framework/TableTreeHelpers.h"
+#include "Framework/Logger.h"
+#include "TFile.h"
+
+#include "ReconstructionDataFormats/PrimaryVertex.h"
+#include "ReconstructionDataFormats/VtxTrackIndex.h"
+#include "ReconstructionDataFormats/VtxTrackRef.h"
+#include "DataFormatsTPC/TrackTPC.h"
+#include "DataFormatsITS/TrackITS.h"
+#include "ReconstructionDataFormats/TrackTPCITS.h"
+
+#include <vector>
+
+using namespace o2;
+using namespace o2::framework;
+
+using GIndex = o2::dataformats::VtxTrackIndex;
+
+template <typename TrackT>
+std::vector<TrackT>* fetchTracks(const char* filename, const char* treename, const char* branchname)
+{
+  TFile file(filename, "OPEN");
+  auto tree = (TTree*)file.Get(treename);
+  auto br = tree->GetBranch(branchname);
+  std::vector<TrackT>* tracks = nullptr;
+  br->SetAddress(&tracks);
+  br->GetEntry(0);
+  return tracks;
+}
```

Comment:
Should not we eventually use the corresponding reader devices? In any case,  I think in the end this will become a workflow sitting in the end of reco. chain.

Reply:
Yes. Will be done later.